var/spack/repos/builtin/packages/babelstream/package.py

```diff
@@ -5,100 +5,217 @@
+        conditional("cmake", when="languages=cxx"),
+        conditional("makefile", when="languages=fortran"),
+        default="cmake",
+    )
+    with when("languages=cxx"):
+        # Also supported variants are cuda and rocm (for HIP)
+        # not included here because they are supplied via respective packages
+        variant("sycl", default=False, description="Enable SYCL support")
+        variant("sycl2020", default=False, description="Enable SYCL support")
+        variant("omp", default=False, description="Enable OpenMP support")
+        variant("ocl", default=False, description="Enable OpenCL support")
+        variant("tbb", default=False, description="Enable TBB support")
+        variant("acc", default=False, description="Enable OpenACC support")
+        variant("hip", default=False, description="Enable HIP support")
+        variant("thrust", default=False, description="Enable THRUST support")
+        variant("raja", default=False, description="Enable RAJA support")
+        variant("std", default=False, description="Enable STD support")
 
     # Some models need to have the programming model abstraction downloaded -
     # this variant enables a path to be provided.
     variant("dir", values=str, default="none", description="Enable Directory support")
+    variant(
+        "sycl2020_submodel",
+        values=("usm", "acc"),
+        when="+sycl2020",
+        default="usm",
+        description="SYCL2020 -> choose between usm and acc methods",
+    )
+    variant(
+        "std_submodel",
+        values=("data", "indices", "ranges"),
+        when="+std",
+        default="data",
+        description="STD -> choose between data, indices and ranges models",
+    )
 
-    # Kokkos conflict and variant
-    conflicts(
-        "dir=none", when="+kokkos", msg="KOKKKOS requires architecture to be specfied by dir="
+    variant(
+        "sycl2020_offload",
+        values=("nvidia", "intel"),
+        default="intel",
+        when="+sycl2020",
+        description="Offloading to NVIDIA GPU or not",
+    )
+
+    variant(
+        "thrust_submodel",
+        values=("cuda", "rocm"),
+        default="cuda",
+        when="+thrust",
+        description="Which THRUST implementation to use, supported options include option= \
+            - CUDA (via https://github.com/NVIDIA/thrust)\
+            - ROCM (via https://github.com/ROCmSoftwarePlatform/rocThrust)",
+    )
+    variant(
+        "thrust_backend",
+        values=("cuda", "omp", "tbb"),
+        default="cuda",
+        when="+thrust",
+        description="Which THRUST implementation to use, supported options include option",
     )
+
+    # Kokkos variant
     variant("kokkos", default=False, description="Enable KOKKOS support")
 
     # ACC conflict
-    variant("cpu_arch", values=str, default="none", description="Enable CPU Target for ACC")
-    variant("acc_target", values=str, default="none", description="Enable CPU Target for ACC")
+
+    variant(
```

Comment:
Couldn't we just use the spec target for this instead of an additional variant? It's a bit weird to say `babelstream cpu_arch=zen3 target=zen4`.

Reply:
Addressed in the new commit [d6400d4](https://github.com/spack/spack/pull/41019/commits/d6400d437dbfd05801f969e43eeb1ef36abbf266)